app/components/contact-page-components/SendMessageForm.tsx

```diff
@@ -0,0 +1,64 @@
+import { RefObject, SetStateAction, FC, FormEvent, Dispatch } from "react";
+
+import FormInput from "./FormInput";
+import Button from "../Button";
+import { inputFields } from "@/constants";
+
+interface SendMessageFormProps {
+  form: RefObject<HTMLFormElement>;
+  sendEmail: (e: FormEvent<HTMLFormElement>) => void;
+  removeErrorMessage: (errorMsg: string) => void;
+  errorList: string;
+  formData: {
+    user_name: string;
+    user_email: string;
+    message: string;
+    contact_info: string;
+  };
+  setFormData: Dispatch<
+    SetStateAction<{
+      user_name: string;
+      user_email: string;
+      message: string;
+      contact_info: string;
```

Comment:
The formData object and the setFormData function share the same type structure. You can simplify this by creating a type alias for the form data structure.

```
type FormData = {
  user_name: string;
  user_email: string;
  message: string;
  contact_info: string;
};
```

```
interface SendMessageFormProps {
  form: RefObject<HTMLFormElement>;
  sendEmail: (e: FormEvent<HTMLFormElement>) => void;
  removeErrorMessage: (errorMsg: string) => void;
  errorList: string;
  formData: FormData;  // 👈 Using FormData here
  setFormData: Dispatch<SetStateAction<FormData>>;  // 👈 And here
}
```
The most important thing is to understand why you're doing this so you can implement it in the future. Don't just replace the code; be sure to understand the "**Why**" when implementing this.